pkg/controller/scylladbdatacenter/resource.go

```diff
@@ -653,15 +663,30 @@ exec /mnt/shared/scylla-operator sidecar \
 --service-name=$(SERVICE_NAME) \
 --cpu-count=$(CPU_COUNT) \
 --loglevel=2 \
-` + func() string {
-										if len(sdc.Spec.ScyllaDB.ExternalSeeds) > 0 {
-											return fmt.Sprintf("--external-seeds=%s", strings.Join(sdc.Spec.ScyllaDB.ExternalSeeds, ","))
-										}
-										return ""
-									}(),
+` +
+										func() string {
+											var optionalArgs []string
+
+											if len(sdc.Spec.ScyllaDB.ExternalSeeds) > 0 {
+												optionalArgs = append(optionalArgs, fmt.Sprintf("--external-seeds=%s", strings.Join(sdc.Spec.ScyllaDB.ExternalSeeds, ",")))
+											}
+
+											return strings.Join(optionalArgs, ` \`)
+										}() +
+										func() string {
+											if len(positionalArgs) > 0 {
+												return ` -- "$@"`
+											}
```

Comment:
nit: I think this can be always set even if the args are empty
(simplifies branching and testing matrix)

Reply:
It would make testing optional args harder, as such test would have to inject the expected arg into string at the right place. Check unit test for external seeds for reference.

Reply:
ok, I tend to prefer code coverage and simpler branching - I don't think unit test complexity should take over it but it's a nit